src/ueye_cam_driver.cpp

```diff
@@ -1174,10 +1174,27 @@ INT UEyeCamDriver::reallocateCamBuffer() {
     return is_err;
   }
 
+  INT width = cam_aoi_.s32Width ;
+  INT height = cam_aoi_.s32Height;
+
+  if((cam_aoi_.s32X > 0) || (cam_aoi_.s32Y > 0))
+  {
+    SENSORINFO m_sInfo;
+    
+	  if((is_err = is_GetSensorInfo(cam_handle_, &m_sInfo)) != IS_SUCCESS) {
+      ERROR_STREAM("Could not retrieve Sensor Info for [" <<
+        cam_name_ << "] (" << err2str(is_err) << ")");
+      return is_err;
+    }
+
+    width = m_sInfo.nMaxWidth;
+    height = m_sInfo.nMaxHeight;
```

Comment:
I am a little confused about this fix. If the AOI X or Y are greater than 0, that should mean that we don't need the full sensor width and height since we are looking at a subset of the image.

Reply:
Indeed we are looking at subset of image and in that sense you are right. However when I look at uEye SDK document, it turns out that memory should be allocated for complete sensor area. Please look at snapshot of document. 
![image](https://user-images.githubusercontent.com/78142297/106453648-ea5a0f80-6489-11eb-8e1e-e16c0110c652.png)

Reply:
The manual states [here](https://de.ids-imaging.com/manuals/ids-software-suite/ueye-manual/4.94/en/is_aoi.html ) that "when changing the size of the AOI, please make sure that the selected image memory is large enough.If it isn't, allocate a new image memory". In combination with the image you posted, which is entitled "Examples for setting absolute AOI positions in memory", I think it would be enough to have the allocated memory just as big as the transmitted image (which can be influenced by width and height but not AOI, right?). 

Nonetheless, you issue clearly states the `IS_INVALID_BUFFER_SIZE` Error. As I have three ETH Cameras wit AOI > 1 working just fine, might this be some race condition? Is the allocated memory maybe not big enough **yet**?

Reply:
Indeed allocated memory isn't big enough. That's why I thought it is wise to allocate memory for full size of sensor. There could be other solution  if we don't want to allocate memory for full size of sensor.
`if(cam_aoi_.s32X > 0)
{
  width = cam_aoi_.s32Width + cam_aoi_.s32X;
}`
`if(cam_aoi_.s32Y > 0)
{
  width = cam_aoi_.s32Height + cam_aoi_.s32Y;
}`
@nullket Could you please try to use camera config file attached in the issue and let see if you can reproduce this problem in your setup.

Reply:
Sorry for the late response: I am not really able to test .ini files right now. But I could test it for you if you would provide a launchfile containing the same parameters? I would then just have to switch the cam id and could report back.